geomesa-fs/geomesa-fs-storage-api/src/main/java/org/locationtech/geomesa/fs/storage/api/PartitionScheme.java

```diff
@@ -0,0 +1,46 @@
+/***********************************************************************
+ * Copyright (c) 2013-2017 Commonwealth Computer Research, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Apache License, Version 2.0
+ * which accompanies this distribution and is available at
+ * http://www.opensource.org/licenses/apache2.0.php.
+ ***********************************************************************/
+
+package org.locationtech.geomesa.fs.storage.api;
+
+import org.opengis.feature.simple.SimpleFeature;
+import org.opengis.feature.simple.SimpleFeatureType;
+import org.opengis.filter.Filter;
+
+public interface PartitionScheme {
+
+    /**
+     * Return the partition in which a SimpleFeature should be stored
+     * @param sf
+     * @return
+     */
+    String getPartitionName(SimpleFeature sf);
```

Comment:
Ok, if you are adding JavaDocs to the interface.... mind doing it for the other classes?